src/components/Calendar.tsx

```diff
@@ -0,0 +1,166 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { ZetkinEvent } from '../types/zetkin';
+import { Heading, Text } from '@adobe/react-spectrum';
+import { useEffect, useRef } from 'react';
+
+interface CalendarProps {
+    focusDate?: Date;
+    events: ZetkinEvent[];
+}
+
+const Calendar = ({ focusDate = new Date(Date.now()), events }: CalendarProps): JSX.Element => {
+    const calendar = useRef<HTMLDivElement>(null);
+    const calendarWrapper = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        const height = calendar.current?.clientHeight || 0;
+        const y = height / 24 * 8; // approx 8am
+        calendarWrapper.current?.scrollTo(0, y);
+    }, []);
+
+    const week = [
+        'misc.calendar.weeks.mon', 'misc.calendar.weeks.tue',
+        'misc.calendar.weeks.wed', 'misc.calendar.weeks.thu',
+        'misc.calendar.weeks.fri', 'misc.calendar.weeks.sat',
+        'misc.calendar.weeks.sun',
+    ];
+    const monday = new Date(new Date(focusDate)
+        .setDate(focusDate.getDate() - focusDate.getDay() + 1));
+    const nextMonday = new Date(new Date(monday)
+        .setDate(monday.getDate() + 7));
+
+    const eventsOfTheWeek = events.filter(event => {
+        return new Date(event.start_time) > monday &&
+            new Date(event.start_time) < nextMonday ||
+            new Date(event.end_time) > monday &&
+            new Date(event.end_time) < nextMonday;
+    });
+
+    const getEventsOfTheDay = (day: number) => {
+        if (day === 7) day = 0; // sunday has index 0 in the Date object
+        return eventsOfTheWeek.filter(event => (
+            new Date(event.start_time).getDay() === day ||
+            new Date(event.end_time).getDay() === day));
+    };
+
+    return (
+        <>
```

Comment:
It's a long component so I might be reading it wrong, but unless I'm mistaken there is only a single top-level element and you should not have to use a [fragment](https://reactjs.org/docs/fragments.html) like this.